packages/overlay/src/integrations/sentry/utils/traces.ts

```diff
@@ -12,17 +12,21 @@ export function groupSpans(spans: Span[]) {
     .sort(a => (a.parent_span_id ? 1 : 0));
 
   sortedSpans.forEach(span => {
-    let parent = idLookup.get(span.parent_span_id || '');
-    span.children = [];
+    let parent = getParentOfSpan(span, idLookup, sortedSpans);
+    log(`parent of span (${span.span_id}, ${span.op}, p: ${span.parent_span_id}) is ${parent?.span_id}`);
+    span.children = span.children || [];
     if (parent) {
-      if (!parent.children) parent.children = [];
+      if (!parent.children) {
+        parent.children = [];
+      }
       parent.children.push(span);
     } else if (span.parent_span_id) {
       const parentParent = sortedSpans.find(s => !s.parent_span_id);
```

Comment:
I think this might be off, too but not sure. Didn't cause problems for this case